app/src/worker.py

```diff
@@ -0,0 +1,106 @@
+import binascii
+import datetime
+import os
+from typing import Optional
+
+from redis import Redis
+
+from common.config import Config
+from common.language import Language
+
+from .app_utils import get_from_and_dkim_domain, scan_and_log
+from .check_results import save_check_results
+from .db import ScanLogEntrySource
+from .logging import build_logger
+from .resolver import setup_resolver
+from .scan import DomainValidationException, ScanningException
+from .translate import translate
+
+LOGGER = build_logger(__name__)
+REDIS = Redis.from_url(Config.Data.REDIS_URL)
+
+setup_resolver()
+
+
+def check_domain_scan_job(
+    client_ip: Optional[str],
+    client_user_agent: Optional[str],
+    envelope_domain: str,
+    token: str,
+    message_key: Optional[bytes] = None,
+    recipient_username: Optional[str] = None,
+) -> None:
+    if message_key:
```

Comment:
imo this function can be split into two, the "has message" and "has no message" parts